### PR TITLE
chore(docs): refactor gh codeblock to use react-query

### PR DESCRIPTION
**Pull request by kevinhalliday**
Refactoring codeblocks to use react query.

task: none

### CODE REVIEW

File: docs/site/docusaurus.config.ts [hunk 1]
```diff
@@ -26,6 +26,8 @@ const config: Config = {
     locales: ["en"],
   },
 
+  clientModules: ["src/client-modules/index.ts"],
```
Comment (idea404) [inline docs/site/docusaurus.config.ts:29]: neat, this does now load before the site

File: docs/site/src/client-modules/react-query-client.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+// prefectch known codeblock queries
+const knownCodeBlocks = [
+  'https://github.com/omni-network/omni/blob/main/contracts/src/interfaces/IOmniPortal.sol',
+  'https://github.com/omni-network/omni/blob/main/contracts/src/pkg/XApp.sol',
+  'https://github.com/omni-network/omni/blob/main/contracts/src/libraries/XTypes.sol',
+  'https://github.com/omni-network/omni-forge-template/blob/main/src/XGreeter.sol',
+  'https://github.com/omni-network/omni/blob/059303647e07fc3481e379b710922e2b84b1827f/contracts/src/pkg/XApp.sol#L56-L65',
+  'https://github.com/omni-network/omni/blob/059303647e07fc3481e379b710922e2b84b1827f/contracts/src/protocol/OmniPortal.sol#L135-L151',
+  'https://github.com/omni-network/omni/blob/059303647e07fc3481e379b710922e2b84b1827f/contracts/src/protocol/OmniPortal.sol#L165-L220',
+]
```
Comment (idea404) [inline docs/site/src/client-modules/react-query-client.ts:15]: Don't think it's easy to maintain adding a `GitHubCodeBlock` in markdown and then adding that same URL here again. Wondering whether we can auto-populate this based on `.md` + `.mdx` files in `./docs`

Comment (kevinhalliday) [reply docs/site/src/client-modules/react-query-client.ts:15]: No it's messy just wanted to show

File: docs/site/src/client-modules/react-query-client.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+const prefech = () =>
+  knownCodeBlocks.forEach(url => client.prefetchQuery(queryOpts(url)))
+
+prefech()
```
Comment (idea404) [inline docs/site/src/client-modules/react-query-client.ts:20]: Cool stuff

File: docs/site/src/client-modules/react-query-client.ts (hunk 2 above)
Comment (kevinhalliday) [reply docs/site/src/client-modules/react-query-client.ts:15]: Also wanted to see if it fixed loading flash on codeblocks, which it does

File: docs/site/src/components/GitHubCodeBlock/useCodeBlock.tsx [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+  const matchesMain = code === codeOnMain
+
+  return {
+    code,
+    matchesMain,
+    mainURL: sourceURL(onMain(file)),
+    sourceURL: sourceURL(file),
+    language: determineLanguage(file.path),
+  }
+}
```
Comment (idea404) [inline docs/site/src/components/GitHubCodeBlock/useCodeBlock.tsx:26]: really like how neat this is too

**Review by idea404 — APPROVED** [review]

**Pull request merged**